src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -1953,6 +1956,12 @@ void G1ConcurrentMark::verify_no_collection_set_oops() {
 #endif // PRODUCT
 
 void G1ConcurrentMark::rebuild_rem_set_concurrently() {
+  // If Remark did not select any regions for RemSet rebuild,
+  // skip the rebuild remembered set phase
+  if (total_selected_for_rebuild() == 0) {
+    log_debug(gc, remset, tracking)("Skipping Remembered Set Rebuild. No Regions Selected For Rebuild.");
```

Comment:
Maybe better is `log_debug(gc, marking)` here, somewhat fitting with the other messages when running with `gc+marking=debug`. Also the upper casing the in the second sentence seems weird, `No regions selected for rebuild` seems better (`Remembered Set Rebuild` is the name of that phase, and we do the upper casing for them already. But the second statement seems to be a regular statement)

Reply:
Thanks, will make both changes